x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanStreamOutput.java

```diff
@@ -80,7 +80,6 @@ public PlanStreamOutput(
     }
 
     public void writeLogicalPlanNode(LogicalPlan logicalPlan) throws IOException {
-        assert logicalPlan.children().size() <= 1;
```

Comment:
Super nit: Maybe this can be left in place?
```java
        assert logicalPlan.children().size() <= 2;
```
or
```java
        assert (logicalPlan.children().size() <= 1) || (logicalPlan instanceOf Join && logicalPlan.children.size() == 2);
```